php/context/class-fieldmanager-context-term.php

```diff
@@ -58,29 +72,75 @@ class Fieldmanager_Context_Term extends Fieldmanager_Context_Storable {
+	 *                             Term" screen? Defaults to yes (true).
+	 *     @type bool $show_on_edit Optional. Should this field show on the
+	 *                              "Edit Term" screen? Defaults to yes (true).
+	 *     @type int $parent Optional. Should this field only show if its parent
+	 *                       matches this term ID?
+	 *     @type bool $use_fm_meta Optional. Should this context store its data
+	 *                             using FM term meta (true, deprecated) or
+	 *                             WordPress core term meta (false). Defaults to
+	 *                             false.
+	 *     @type Fieldmanager_Field $field Optional. The field to which to
+	 *                                     attach this context.
+	 * }
+	 * @param string|array $taxonomies Optional. Deprecated. Required if $args
+	 *                                 is a string.
+	 * @param boolean $show_on_add Optional. Deprecated.
+	 * @param boolean $show_on_edit Optional. Deprecated.
+	 * @param string $parent Optional. Deprecated.
+	 * @param Fieldmanager_Field $fm Optional. Deprecated.
 	 */
-	public function __construct( $title, $taxonomies, $show_on_add = true, $show_on_edit = true, $parent = '', $fm = null ) {
-		// Populate the list of taxonomies for which to add this meta box with the given settings
-		if ( ! is_array( $taxonomies ) ) {
-			$taxonomies = array( $taxonomies );
-		}
+	public function __construct( $args, $taxonomies = array(), $show_on_add = true, $show_on_edit = true, $parent = '', $fm = null ) {
+		if ( is_array( $args ) ) {
+			$args = wp_parse_args( $args, array(
+				'show_on_add'  => true,
+				'show_on_edit' => true,
+				'parent'       => '',
+				'use_fm_meta'  => false,
+				'field'        => null,
+			) );
+			if ( ! isset( $args['title'], $args['taxonomies'] ) ) {
+				throw new FM_Developer_Exception( esc_html__( '"title" and "taxonomies" are required values for Fieldmanager_Context_Term', 'fieldmanager' ) );
+			}
 
-		// Set the class variables
-		$this->title = $title;
-		$this->taxonomies = $taxonomies;
-		$this->show_on_add = $show_on_add;
-		$this->show_on_edit = $show_on_edit;
-		$this->parent = $parent;
-		$this->fm = $fm;
+			$this->title        = $args['title'];
+			$this->taxonomies   = (array) $args['taxonomies'];
+			$this->show_on_add  = $args['show_on_add'];
+			$this->show_on_edit = $args['show_on_edit'];
+			$this->parent       = $args['parent'];
+			$this->use_fm_meta  = $args['use_fm_meta'];
+			$this->fm           = $args['field'];
+		} elseif ( empty( $taxonomies ) ) {
+			throw new FM_Developer_Exception( esc_html__( '"title" and "taxonomies" are required values for Fieldmanager_Context_Term', 'fieldmanager' ) );
+		} else {
+			_deprecated_argument( __METHOD__, 'Fieldmanager-1.0.0-beta.3', esc_html__( 'Instantiating Fieldmanager_Context_Term using individual arguments is deprecated. Please pass an array of arguments instead.', 'fieldmanager' ) );
```

Comment:
Again, I think it's fine to support two approaches for passing arguments, but I don't think it makes sense to immediately start calling `_deprecated_*()`